modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceDetailsViewManager.java

```diff
@@ -298,5 +331,133 @@ private ServiceQueryModel getServiceQueryModel(Category category, Csp csp, Strin
+        if (serviceDeployment.getServiceDeploymentState()
+                .equals(ServiceDeploymentState.DEPLOYING)) {
+            serviceOrderEntity = serviceOrderEntities
+                    .stream()
+                    .filter(serviceOrder ->
+                            serviceOrder.getTaskType().equals(ServiceOrderType.DEPLOY)).findFirst()
+                    .orElseThrow(
+                            () -> new ServiceOrderNotFound(
+                                    "No ServiceOrderEntity found with ServiceOrderType DEPLOY"));
+        } else if (serviceDeployment.getServiceDeploymentState()
+                .equals(ServiceDeploymentState.DESTROYING)) {
+            serviceOrderEntity = serviceOrderEntities
+                    .stream()
+                    .filter(serviceOrder ->
+                            serviceOrder.getTaskType().equals(ServiceOrderType.DESTROY)).findFirst()
+                    .orElseThrow(() -> new ServiceOrderNotFound(
+                            "No ServiceOrderEntity found with ServiceOrderType DESTROY"));
+        } else if (serviceDeployment.getServiceDeploymentState()
+                .equals(ServiceDeploymentState.MODIFYING)) {
+            serviceOrderEntity = serviceOrderEntities
+                    .stream()
+                    .filter(serviceOrder ->
+                            serviceOrder.getTaskType().equals(ServiceOrderType.MODIFY)).findFirst()
+                    .orElseThrow(
+                            () -> new ServiceOrderNotFound(
+                                    "No ServiceOrderEntity found with ServiceOrderType MODIFY"));
+        } else {
+            return serviceOrderEntity;
+        }
+        return serviceOrderEntity;
+    }
+
+    private void updateServiceDeploymentStateAndServiceOrder(
+            ServiceDeploymentEntity serviceDeployment,
+            ServiceOrderEntity serviceOrderEntity, ErrorType errorType,
+            TerraformBootRequestFailedException e) {
+        serviceDeployment.setServiceDeploymentState(ServiceDeploymentState.MANUAL_CLEANUP_REQUIRED);
+        serviceOrderEntity.setTaskStatus(TaskStatus.FAILED);
+        serviceOrderEntity.setErrorResponse(ErrorResponse.errorResponse(
+                errorType, List.of(e.getMessage())));
+        serviceDeploymentStorage.storeAndFlush(serviceDeployment);
+        databaseServiceOrderStorage.storeAndFlush(serviceOrderEntity);
+    }
+
+    private void updateServiceDeploymentState(ServiceDeploymentEntity serviceDeployment,
+            ServiceOrderEntity serviceOrderEntity) {
+        if (Objects.nonNull(serviceOrderEntity)) {
+            if (Duration.between(serviceOrderEntity.getStartedTime(),
+                    OffsetDateTime.now()).getSeconds() > maxServiceOrderProcessingDuration) {
+                if (serviceOrderEntity.getHandler().equals(Handler.TERRAFORM_BOOT)) {
+                    try {
+                        ResponseEntity<TerraformResult> result = terraformBootTaskResultHelper
```

Comment:
would prefer if we extract the response object in the terraformBootTaskResultHelper class itself and send the actual response object